src/charm.py

```diff
@@ -129,9 +128,22 @@ def _on_update_status(self, _):
     # END: hook functions
 
     # BEGIN: helper functions
+    def update_external_services(self) -> None:
+        """Attempts to update any external Kubernetes services."""
+        if not self.is_external_client:
+            return
+
+        # every unit attempts to create a nodeport service
+        # if exists, will silently continue
+        self.node_port_manager.apply_service(
+            service=self.node_port_manager.build_node_port_services(
+                port=Config.MONGOS_PORT
+            )
+        )
```

Comment:
Sorry I don't follow 🙈 are you suggesting I merge apply_service and build_node_port_services into one function?

Reply:
I'd personally rather keep it modular and similar to Kafka's for maintainability.

Mostly because I am worried about changes being needed down to road and having a hard time porting over changes from one to another. I am going to keep it for now. I hope thats okay!